aws-s3/scripts-ccao-data-raw-us-east-1/spatial-point_source_air_noise.R

```diff
@@ -0,0 +1,168 @@
+
+#treat centroid of each airport as point source of noise
+ohare_point <- c(site = "ohare", 
+                 latitude = 41.97857577880779, 
+                 longitude = -87.90817373313197)
+midway_point <- c(site = "midway", 
+                  latitude= 41.78512649107475, 
+                  longitude = -87.75182050036706)
+aps <- as.data.frame(rbind(ohare_point, midway_point)) %>%
+  st_as_sf(coords = c("longitude", "latitude")) %>% 
+  st_set_crs(4326) %>%
+  st_transform(3435)
+
+#get distance in feet between each sensor and both airports
+sensor_dists <- st_distance(airport_clean, aps)
+
+airport_clean$distance_ohare <- sensor_dists[,1]
+airport_clean$distance_midway <- sensor_dists[,2]
+
+# Modeling choice: assume a baseline DNL in dB of 50, 
+# per FAA info about average for "quiet, suburban, residential" environment
+# https://www.faa.gov/regulations_policies/policy_guidance/noise/community
+BASELINE_DNL <- 50
+
+# Modeling choice: assume ALL noise above baseline DNL for a sensor is from the 
+# airport the sensor is closest to.
+
+# Intensity of sound falls off with the inverse square of distance from point source
+# I = Ax^-2, where A represents W/4pi: 
+# https://blog.soton.ac.uk/soundwaves/wave-basics/point-sources-inverse-square-law/
+
+# To model this falloff, we convert decibels to absolute intensity (since
+# dB is logarithmic), subtract our baseline, and then do a linear regression with 
+# intensity as y and inverse square of distance as x.
+# Our coefficient minimizes sum of squared error for the 2011-19 sensor averages
+
+# transform distance variable into inverse distance-squared
+airport_clean$isq_dist_ohare <- (airport_clean$distance_ohare)^-2
+airport_clean$isq_dist_midway <- (airport_clean$distance_midway)^-2
+
+## Interconversion functions for dB and intensity
+
+I_ZERO = 1e-12
+
+# https://courses.lumenlearning.com/atd-austincc-physics1/chapter/17-3-sound-intensity-and-sound-level/
+decibel_to_intensity <- function(decibel){
+  return(10^(decibel/10) * I_ZERO)
+}
+
+intensity_to_decibel <- function(I){
+  return (10 * log10(I/I_ZERO))
+}
+
+airport_clean$total_intensity <- decibel_to_intensity(airport_clean$noise)
+airport_clean$added_intensity <- decibel_to_intensity(airport_clean$noise - BASELINE_DNL)
+
+# Verify against https://www.omnicalculator.com/physics/db 
+# (change "SIL" entry, leave rest same)
+# for (i in 1:150){
+#   print(paste0("Decibel: ", i, " Intensity: ", decibel_to_intensity(i)))
+# }
+
+## PREDICT AIRPORT NOISE LEVELS ================================================
+
+# Set intercept at 0, i.e., when linear distance from airport is infinite,
+# the absolute intensity of sound emanating from it will be 0
+# (Decibel measure will NOT drop to 0; it can keep going to -Inf since 0 dB is 
+# just a a cutoff of human *perception* of sound)
+
+ohare_model_intensity <- lm(added_intensity ~ 0 + isq_dist_ohare, data = airport_clean %>% filter(airport == "ohare")) 
+# Adjusted R-squared:  0.4002 
```

Comment:
**question:** This regression is minimizing squared errors for the average of _all_ noise readings, 2011 - 2019, correct? 

If so, I have two thoughts/suggestions:

1. Make sure you're using the OMP (O'Hare Modernization Project) values for O'Hare, as those are the "current" set of monitor results.
2. I expect that the noise intensity has changed over time. Perhaps we should include `year` as an independent variable in the regression instead of using the average. This complicates the modeling and feature construction a little bit but would probably produce a better fit.